config/config.go

```diff
@@ -293,6 +294,51 @@ func GetTransport() *http.Transport {
 	return transport
 }
 
+func InitConfig() {
+	// 1) Set up defaults.yaml
+	err := viper.MergeConfig(strings.NewReader(defaultsYaml))
+	if err != nil {
+		cobra.CheckErr(err)
+	}
+	// 2) Set up osdf.yaml (if needed)
+	prefix := GetPreferredPrefix()
+	if prefix == "OSDF" {
+		err := viper.MergeConfig(strings.NewReader(osdfDefaultsYaml))
+		if err != nil {
+			cobra.CheckErr(err)
+		}
+	}
+	if viper.GetString("config") != "" {
+		viper.SetConfigFile(viper.GetString("config"))
+	} else {
+		home, err := os.UserHomeDir()
+		if err != nil {
+			log.Warningln("No home directory found for user -- will check for configuration yaml in /etc/pelican/")
+		}
+
+		// 3) Set up pelican.yaml (has higher precedence)
+		viper.AddConfigPath(filepath.Join(home, ".config", "pelican"))
+		viper.AddConfigPath(filepath.Join("/etc", "pelican"))
+		viper.SetConfigType("yaml")
+		viper.SetConfigName("pelican")
+	}
+
+	viper.SetEnvPrefix(GetPreferredPrefix())
```

Comment:
We can avoid an extra call to `GetPreferredPrefix` here --> we already have the value stored in a variable called `prefix`